### ISSUE TITLE
Slot Game Name and $ Overlap

### ISSUE THREAD

**Issue by Grengras** [JJM152/Queen#84]
As subject states. In the slot game often a long name such as Captain Reginald will overlap the name text with the $$ symbols telling the player the value of that choice. This happens with even some random pirates with just a single $ mark if they have exceptionally long names. For example I am right now staring at "Pirate Grundemore" who has the last "e" of his name completely overlapped / overlapping with a single "$" making it impossible to read either symbol without left click and drag highlighting. Dragging left to right highlights the "re" name ending, while dragging right to left highlights the "$".

List of effected names includes the pirate captain (just barely) and quartermaster (extremely) of the ship, the guard captain (heavily), any random name longer than 8 letters when a title like "Pirate" gets added in front.

Possible fixes would be either:

1) Change the max length of these names so they are all shorter than the $ symbols. (Probably the easiest fix because don't need to change anything visually / code wise. However the max length that fits is only 15 characters including spaces and titles which is tough to fit with long titles like "Pirate" meaning need to limit random names to just 8 letters, and the captain's title in slots needs to be dropped to Cptn instead of full, the quartermaster needs slot title to become just QM, and the captain of the guard either needs slot title changed to just CoG or else his actual name needs to be made shorter.); or

2) Move the $ values to front and name displays second on line. This would make the long names end abruptly at the end of their box but would avoid overlap with the money value.; or

3) Change the $ to $$$$$ system instead to simply a number value 1-5 followed by a single $.; or

4) As 3 except only the $ shows up without any number, but the color of the symbol changes depending on its worth. Red for 1, orange for 2, yellow for 3, green for 4, blue for 5. (Will this system work for color blind people?); or

5) As 4 except instead of using colors it uses font formatting. 1 would be normal, 2 would be italicized, 3 would be bold, 4 would be both italicized and bold, 5 would be italicized and bold and underlined. (Would these differences be noticeable enough for players to understand intuitively the differences?)

**Comment by JJM152** [comment]
I investigated this some more and I can't reproduce it. 

I suspect that it may be related to your browser. If you can send me the name of the browser you are using, what version it is, and attach a screenshot to this issue, maybe I can help. 

I say that, but on some level I don't want to get too into tweaking stuff for different browsers unless it's literally unplayable.

**Comment by Grengras** [comment]
I use Firefox (version 61.0.2 64-bit)

As for screenshots... 
![screenhunter_01 oct 07 00 24](https://user-images.githubusercontent.com/14894043/46578220-dee49f80-c9c7-11e8-8e1d-a1f30d25e1a2.jpg) shows QuarterMaster Julius in top left corner, or should that be Juliu$$$ since the s is overlapped by the first $.

Next we have, 
![screenhunter_02 oct 07 00 35](https://user-images.githubusercontent.com/14894043/46578255-0d16af00-c9c9-11e8-8732-9a5aeb561f95.jpg) which shows both what can happen with a long named pirate and with the Captain of the Ship.

It is certainly by no means unplayable, and if I didn't have a mild case of OCD I would probably not even mention it. But, yeah, it is most definitely noticeable.

As an aside, I find that if I change the browser Zoom level it changes slightly the readability a bit, but it remains visible at all zoom levels except 90%. 80% or lower and 100% or higher all scrunch up the text like that for me. And even at 90% the text of the first $ remains very close.

**Comment by JJM152** [comment]
It looks to me like it's an issue with the text size/font. My testing shows that your browser isn't respecting the 'smaller' font-size attribute for some reason.

![image](https://user-images.githubusercontent.com/8572919/46579185-c15b1880-ca4f-11e8-984c-9a7cbd7ec8fc.png)

[This stackflow page](https://stackoverflow.com/questions/36524272/why-does-firefox-not-honor-css-font-size-for-code-tags-wrapped-in-pre) suggests that it's an issue with Firefox rendering monospace fonts and that the fix is to change `font-family: monospace` to` font-family:monospace, monospace`.

I've done that on my local copy and it didn't break anything in Chrome, but I don't have firefox installed to test. If you want to make that edit on your local copy and try it out, let me know, otherwise I'll test it when I get around to it.

Cheers.

**Comment by Grengras** [comment]
I've been tweeking this back and forth trying to see what worked for me and what didn't. I changed monospace; to monospace, monospace; it for some reason messed up the status bar to the right also, although it did fix the issue I saw.

So the next thing I tried was to see if I could fix the overlap of $ without also breaking the whorestatus on the right side. Finally I found something that worked. Changing the left pixels tag in the div.WhoreCustomerGUI function. If you only edit this one div statement the overlap is fixed without any problems. Simply switch "monotone;" with "monotone,;" and it perfectly fixes it. Who knew the power of the simple comma? O_o

div.WhoreCustomerGUI {
    position: absolute;
    top: 5px;
    left: 5px;
    width: 180px;
    height: 130px;
    border-radius: 5px 25px;
    border: 1px solid saddlebrown;
    font-family: monospace,;
    cursor: pointer;
}
